File: java-example/src/test/java/org/onflow/examples/java/signTransaction/SignTransactionExampleTest.java
package org.onflow.examples.java.signTransaction;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.onflow.flow.common.test.*;
import org.onflow.flow.sdk.*;

import java.util.List;

@FlowEmulatorProjectTest(flowJsonLocation = "../flow/flow.json")
public class SignTransactionExampleTest {
    @FlowServiceAccountCredentials
    private TestAccount serviceAccount;

    @FlowTestAccount
    private TestAccount testAccount;

    @FlowTestAccount
    private TestAccount testAccount2;

    @FlowTestClient
    private FlowAccessApi accessAPI;

    private SignTransactionExample transactionConnector;

    @BeforeEach
    public void setup() {
        transactionConnector = new SignTransactionExample(serviceAccount.getPrivateKey(), accessAPI);
    }

    @Test
    public void canSignSinglePartySingleSigTransaction() throws Exception {
        FlowTransactionResult txResult = transactionConnector.singlePartySingleSignature(serviceAccount.getFlowAddress(), "cadence/simple_transaction.cdc", 500L);

        Assertions.assertNotNull(txResult);
        Assertions.assertSame(txResult.getStatus(), FlowTransactionStatus.SEALED, "Transaction should be sealed");
    }

    @Test
    public void canSignSinglePartyMultiSigTransaction() throws Exception {
        transactionConnector = new SignTransactionExample(testAccount.getPrivateKey(), accessAPI);
        FlowTransactionResult txResult = transactionConnector.singlePartyMultiSignature(testAccount.getFlowAddress(), "cadence/simple_transaction.cdc", 500L);

        Assertions.assertNotNull(txResult);
        Assertions.assertSame(txResult.getStatus(), FlowTransactionStatus.SEALED, "Transaction should be sealed");
    }

    @Test
    public void canSignMultiPartySingleSigTransaction() throws Exception {
        FlowTransactionResult txResult = transactionConnector.multiPartySingleSignature(
                serviceAccount.getPrivateKey(),
                testAccount.getPrivateKey(),
                testAccount.getFlowAddress(),
                serviceAccount.getFlowAddress(),
                "cadence/simple_transaction.cdc",
                500L
        );

        Assertions.assertNotNull(txResult);
        Assertions.assertSame(txResult.getStatus(), FlowTransactionStatus.SEALED, "Transaction should be sealed");
    }

    @Test
    public void canSignMultiPartyMultiSigTransaction() throws Exception {
        FlowTransactionResult txResult = transactionConnector.multiPartyMultiSignature(
                List.of(testAccount2.getPrivateKey(), testAccount2.getPrivateKey()),
                List.of(testAccount.getPrivateKey(), testAccount.getPrivateKey()),
                testAccount.getFlowAddress(),
                testAccount2.getFlowAddress(),
                "cadence/simple_transaction.cdc",
                500L
        );

        Assertions.assertNotNull(txResult);
        Assertions.assertSame(txResult.getStatus(), FlowTransactionStatus.SEALED, "Transaction should be sealed");
    }

    @Test
    public void canSignMultiParty2AuthorizersTransaction() throws Exception {
        FlowTransactionResult txResult = transactionConnector.multiParty2Authorizers(
                serviceAccount.getPrivateKey(),
                testAccount.getPrivateKey(),
                testAccount.getFlowAddress(),
                serviceAccount.getFlowAddress(),
                testAccount.getFlowAddress(),
                "cadence/simple_transaction_2_authorizers.cdc",
                500L
        );

        Assertions.assertNotNull(txResult);
        Assertions.assertSame(txResult.getStatus(), FlowTransactionStatus.SEALED, "Transaction should be sealed");
    }
}

File: kotlin-example/src/test/kotlin/org/onflow/examples/kotlin/signTransaction/SignTransactionExampleTest.kt
package org.onflow.examples.kotlin.signTransaction

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.onflow.flow.common.test.*
import org.onflow.flow.sdk.*

@FlowEmulatorProjectTest(flowJsonLocation = "../flow/flow.json")
internal class SignTransactionExampleTest {
    @FlowServiceAccountCredentials
    lateinit var serviceAccount: TestAccount

    @FlowTestAccount
    lateinit var testAccount: TestAccount

    @FlowTestAccount
    lateinit var testAccount2: TestAccount

    @FlowTestClient
    lateinit var accessAPI: FlowAccessApi

    private lateinit var transactionConnector: SignTransactionExample

    @BeforeEach
    fun setup() {
        transactionConnector = SignTransactionExample(serviceAccount.privateKey, accessAPI)
    }

    @Test
    fun `Can sign single party single sig transaction`() {
        val txResult = transactionConnector.singlePartySingleSignature(serviceAccount.flowAddress)

        Assertions.assertNotNull(txResult)
        Assertions.assertTrue(txResult.status === FlowTransactionStatus.SEALED, "Transaction should be sealed")
    }

    @Test
    fun `Can sign single party multi-sig transaction`() {
        transactionConnector = SignTransactionExample(testAccount.privateKey, accessAPI)
        val txResult = transactionConnector.singlePartyMultiSignature(testAccount.flowAddress)

        Assertions.assertNotNull(txResult)
        Assertions.assertTrue(txResult.status === FlowTransactionStatus.SEALED, "Transaction should be sealed")
    }

    @Test
    fun `Can sign multi-party single sig transaction`() {
        val txResult = transactionConnector.multiPartySingleSignature(
            account1PrivateKey = serviceAccount.privateKey,
            account2PrivateKey = testAccount.privateKey,
            payerAddress = testAccount.flowAddress,
            authorizerAddress = serviceAccount.flowAddress
        )

        Assertions.assertNotNull(txResult)
        Assertions.assertTrue(txResult.status === FlowTransactionStatus.SEALED, "Transaction should be sealed")
    }

    @Test
    fun `Can sign multi-party multi-sig transaction`() {
        val txResult = transactionConnector.multiPartyMultiSignature(
            account1PrivateKeys = listOf(testAccount2.privateKey, testAccount2.privateKey),
            account2PrivateKeys = listOf(testAccount.privateKey, testAccount.privateKey),
            payerAddress = testAccount.flowAddress,
            authorizerAddress = testAccount2.flowAddress
        )

        Assertions.assertNotNull(txResult)
        Assertions.assertTrue(txResult.status === FlowTransactionStatus.SEALED, "Transaction should be sealed")
    }

    @Test
    fun `Can sign multi-party 2 authorizers transaction`() {
        val txResult = transactionConnector.multiParty2Authorizers(
            account1PrivateKey = serviceAccount.privateKey,
            account2PrivateKey = testAccount.privateKey,
            payerAddress = testAccount.flowAddress,
            authorizer1Address = serviceAccount.flowAddress,
            authorizer2Address = testAccount.flowAddress
        )

        Assertions.assertNotNull(txResult)
        Assertions.assertTrue(txResult.status === FlowTransactionStatus.SEALED, "Transaction should be sealed")
    }
}
